test/AutoDiff/SILOptimizer/pullback_generation.sil

```diff
@@ -0,0 +1,155 @@
+// Pullback generation tests written in SIL for features 
+// that may not be directly supported by the Swift frontend
+
+// RUN: %target-sil-opt --differentiation -debug-only=differentiation -emit-sorted-sil %s 2>&1 | %FileCheck %s
+
+//===----------------------------------------------------------------------===//
+// Pullback generation - `struct_extract`
+// - Input to pullback has non-owned ownership semantics which requires copying
+// this value to stack before lifetime-ending uses. 
+//===----------------------------------------------------------------------===//
+
+sil_stage raw
+
+import Builtin
+import Swift
+import SwiftShims
+
+import _Differentiation
+
+struct X {
+  @_hasStorage var a: Float { get set }
+  @_hasStorage var b: String { get set }
+  init(a: Float, b: String)
+}
+
+extension X : Differentiable, Equatable, AdditiveArithmetic {
+  public typealias TangentVector = X
+  mutating func move(by offset: X)
+  public static var zero: X { get }
+  public static func + (lhs: X, rhs: X) -> X
+  public static func - (lhs: X, rhs: X) -> X
+  @_implements(Equatable, ==(_:_:)) static func __derived_struct_equals(_ a: X, _ b: X) -> Bool
+}
+
+struct Y {
+  @_hasStorage var a: X { get set }
+  @_hasStorage var b: String { get set }
+  init(a: X, b: String)
+}
+
+extension Y : Differentiable, Equatable, AdditiveArithmetic {
+  public typealias TangentVector = Y
+  mutating func move(by offset: Y)
+  public static var zero: Y { get }
+  public static func + (lhs: Y, rhs: Y) -> Y
+  public static func - (lhs: Y, rhs: Y) -> Y
+  @_implements(Equatable, ==(_:_:)) static func __derived_struct_equals(_ a: Y, _ b: Y) -> Bool
+}
+
+sil_differentiability_witness hidden [reverse] [parameters 0] [results 0] @$function_with_struct_extract_1 : $@convention(thin) (@guaranteed Y) -> @owned X {
+}
+
+sil hidden [ossa] @$function_with_struct_extract_1 : $@convention(thin) (@guaranteed Y) -> @owned X {
+bb0(%0 : @guaranteed $Y):
+  %1 = struct_extract %0 : $Y, #Y.a               
+  %2 = copy_value %1 : $X                         
+  return %2 : $X                                  
+}
+
+// CHECK-LABEL: [ORIG]  %1 = struct_extract %0 : $Y, #Y.a               // user: %2
```

Comment:
I would rather do `CHECK-LABEL` for function name + see comment below

Reply:
Will change this to use the actual SIL after the differentiation pass instead.